jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildTarballAndLoadDockerStep.java

```diff
@@ -0,0 +1,149 @@
+    return Futures.whenAllComplete(dependencies)
+        .call(this::afterPushBaseImageLayerFuturesFuture, listeningExecutorService);
+  }
+
+  /**
+   * Depends on {@code pullBaseImageLayerFuturesFuture.get()} and (@code
+   * buildConfigurationFutureFuture.get()}.
+   *
+   * <p>TODO: Refactor into testable components
+   */
+  private Void afterPushBaseImageLayerFuturesFuture()
+      throws ExecutionException, InterruptedException, IOException {
+    // Add layers to image tarball
+    TarStreamBuilder tarStreamBuilder = new TarStreamBuilder();
+    List<String> layerFiles = new ArrayList<>();
+    for (Future<CachedLayer> cachedLayerFuture :
+        NonBlockingFutures.get(pullBaseImageLayerFuturesFuture)) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      tarStreamBuilder.addEntry(
+          new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+    for (Future<CachedLayer> cachedLayerFuture : buildApplicationLayerFutures) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      tarStreamBuilder.addEntry(
+          new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+
+    // Add config to tarball
+    // TODO: Add ability to add blobs as entries to TarStreamBuilder without using temp files
+    Blob containerConfigurationBlob =
+        NonBlockingFutures.get(NonBlockingFutures.get(buildConfigurationFutureFuture));
+    Path tempConfig = Files.createTempFile(null, null);
+    tempConfig.toFile().deleteOnExit();
+    try (OutputStream bufferedOutputStream =
+        new BufferedOutputStream(Files.newOutputStream(tempConfig))) {
+      containerConfigurationBlob.writeTo(bufferedOutputStream);
+    }
+    tarStreamBuilder.addEntry(new TarArchiveEntry(tempConfig.toFile(), "config.json"));
+
+    // Add manifest to tarball
+    Blob manifestBlob = getManifestBlob(buildConfiguration.getTargetImageReference(), layerFiles);
+    Path tempManifest = Files.createTempFile(null, null);
+    tempManifest.toFile().deleteOnExit();
+    try (OutputStream bufferedOutputStream =
+        new BufferedOutputStream(Files.newOutputStream(tempManifest))) {
+      manifestBlob.writeTo(bufferedOutputStream);
+    }
+    tarStreamBuilder.addEntry(new TarArchiveEntry(tempManifest.toFile(), "manifest.json"));
+
+    // Load the image to docker daemon
+    // TODO: Command is untested/not very robust
+    new Command("docker", "load").run(Blobs.writeToByteArray(tarStreamBuilder.toBlob()));
+
+    return null;
+  }
+
+  /**
+   * Builds a {@link DockerLoadManifestTemplate} from image parameters and returns the result as a
+   * blob.
+   */
+  @VisibleForTesting
+  static Blob getManifestBlob(ImageReference imageReference, List<String> layerFiles) {
```

Comment:
For this project, we're following the convention of static methods at the top of a class

The ordering is:

1. Constants
1. static fields
1. static methods
1. member fields
1. constructors
1. instance methods

Reply:
Above the private member fields too.